File: merkle-trie/src/main/java/net/consensys/cava/trie/MerkleStorageException.java
package net.consensys.cava.trie;

/**
 * This exception is thrown when there is an issue retrieving or decoding values from {@link MerkleStorage}.
 */
public class MerkleStorageException extends RuntimeException {

  public MerkleStorageException(String message) {
    super(message);
  }

  public MerkleStorageException(String message, Exception cause) {
    super(message, cause);
  }
}

File: merkle-trie/src/main/kotlin/net/consensys/cava/trie/experimental/MerkleStorageException.kt
package net.consensys.cava.trie.experimental

/**
 * This exception is thrown when there is an issue retrieving or decoding values from [MerkleStorage].
 */
class MerkleStorageException : RuntimeException {

  constructor(message: String) : super(message)

  constructor(message: String, cause: Exception) : super(message, cause)
}
